src/main/java/io/phasetwo/service/resource/UserResourceProviderFactory.java

```diff
@@ -24,7 +42,72 @@ public RealmResourceProvider create(KeycloakSession session) {
   public void init(Config.Scope config) {}
 
   @Override
-  public void postInit(KeycloakSessionFactory factory) {}
+  public void postInit(KeycloakSessionFactory factory) {
+
+    log.debug("UserResourceProviderFactory::postInit");
+
+    factory.register(
+        (ProviderEvent event) -> {
+          if (event instanceof RealmModel.RealmPostCreateEvent) {
+            log.debug("RealmPostCreateEvent");
+            realmPostCreateInitUserProfile((RealmModel.RealmPostCreateEvent) event);
+          } else if (event instanceof PostMigrationEvent) {
+            log.debug("PostMigrationEvent");
+            KeycloakModelUtils.runJobInTransaction(factory, this::postMigrationInitUserProfile);
```

Comment:
Want this to be conditional on the `KC_ORGS_SKIP_MIGRATION` env var. The reason we need to be able to skip this is that sometimes installations with very large numbers of realms can fail because of large transactions (think 1000's of realm updates in a single transaction.

See https://github.com/p2-inc/keycloak-orgs/blob/main/src/main/java/io/phasetwo/service/resource/OrganizationResourceProviderFactory.java#L64

Perhaps we can make this a little more structured than reading the env var everywhere.

Reply:
Thanks, I wasn't aware of this possibility, I have updated it